common/persistence/visibility/store/sql/visibility_store.go

```diff
@@ -471,7 +471,12 @@ func (s *VisibilityStore) processRowSearchAttributes(
 	for name, value := range rowSearchAttributes {
 		tp, err := saTypeMap.GetType(name)
 		if err != nil {
-			return nil, err
+			// TODO: CHASM search attributes are not in the typeMap and SQL only stores raw values (no metadata).
+			// The Encode() call below will fail to add type metadata, causing decode issues.
+			if !searchattribute.IsChasmSearchAttribute(name) {
+				return nil, err
+			}
+			continue
 		}
```

Comment:
Wouldn't this make more sense?


```suggestion
		// TODO: CHASM search attributes are not in the typeMap and SQL only stores raw values (no metadata).
		// The Encode() call below will fail to add type metadata, causing decode issues.
		if searchattribute.IsChasmSearchAttribute(name) {
			continue
		}
		tp, err := saTypeMap.GetType(name)
		if err != nil {
			return nil, err
		}
```